### PR TITLE
Chore: Adds styles layers to help with CSS specificity issues

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Adds styles layers to deal with CSS specificity issues

## How it works?

We face a CSS specificity issue when using the overriding component and creating a custom component using the FS UI component. We are setting an order for customization CSS specificity through css cascade layers.

|Before|
|-|
|image|

Now the priority should be:
`Users custom styles**` > `theme` > `components (default ui styles)`

** un-layered styles have the highest priority

|w/ css layers|
|-|
|image|

The cascade layers feature seems to have good coverage (91.23%) support by the browsers. So we've decided it would be a practical approach for this issue.

## How to test it?

You can go through this [guide](https://faststore-site-git-chore-adds-styles-layers-fs-1185-faststore.vercel.app/docs/customizing/components#customizing-a-new-component-when-using-override), and play with the custom stylesheet. Check if the custom stylesheet has a higher priority when setting your styles.

1. Try it with the current starter repo version (You won't see changes in the button's borders nor padding)
2. Then try on this [starter branch](https://github.com/vtex-sites/starter.store/pull/114).  (You will be able to see all the changes applied)

## References

- [Cascade Layers](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_layers)
- https://caniuse.com/css-cascade-layers

### CODE REVIEW

File: apps/site/pages/docs/themes/overview.mdx [hunk 1]
```diff
@@ -46,37 +46,40 @@ By default, every FastStore project comes with the [Brandless theme](/docs/themi
...
-  // --------------------------------------------------------
-
-  // --------------------------------------------------------
-  // FS UI Components
-  // --------------------------------------------------------
-    .section {
-      // Add here the customizations for FastStore UI components.
-    }
+
+@layer theme {
```
Comment (hellofanny) [inline apps/site/pages/docs/themes/overview.mdx:50]: This was the easiest way that I thougth to add this layer to the priority list.
Does anyone have a better idea, given our current structure?

Comment (hellofanny) [reply apps/site/pages/docs/themes/overview.mdx:50]: I still need to update this on our documentation!

File: apps/site/pages/docs/customizing/components.mdx [hunk 2]
```diff
@@ -117,6 +117,59 @@ Notice that in this example, only one section received the change, the `ProductS
...
+
+```scss filename="CustomBuyButton.module.scss"
+.CustomBuyButton {
+  --fs-button-border-radius: var(--fs-border-radius-pill);
+  --fs-button-padding:  0 var(--fs-spacing-5);
+  color: var(--fs-color-text-inverse);
+  [data-fs-button-wrapper] {
+    background-color: #E31C58;
+  }
+}
```
Comment (renatamottam) [inline apps/site/pages/docs/customizing/components.mdx:157]: just a suggestion :)
```suggestion
.CustomBuyButton {
  --fs-button-border-radius: var(--fs-border-radius-pill);
  --fs-button-padding:  0 var(--fs-spacing-5);
  
  color: var(--fs-color-text-inverse);
  
  [data-fs-button-wrapper] {
    background-color: #E31C58;
  }
}
```

**Review by renatamottam — APPROVED** [review]
This is incredible! Thanks for this huge improvement to the product, Fan, you rock 🤘

**Review by icazevedo — APPROVED** [review]
LGTM. Warn clients about the `@layer theme` change when you merge it.

**Pull request merged**